### ISSUE TITLE
Onnx Test Code

### ISSUE THREAD

**Issue by hghari** [ultralytics/yolov5#327]
Hello, Thanks for the great work. I have exported onnx but dont know how to use the output which is 1 by 3 by 80 by 80 by 15 tensor. Sounds like this is one of the three outputs. Why are you excluding the last layer in exporting to Onnx and how we have to calculate the results having these three branches of 1x3x80x80x15, 1x3x20x20x15 and 1x3x40x40x15. I have ten classes. I guess somehow I have to add this part myself to the onnx model to make predictions:

            if not self.training:  # inference
                if self.grid[i].shape[2:4] != x[i].shape[2:4]:
                    self.grid[i] = self._make_grid(nx, ny).to(x[i].device)
                y = x[i].sigmoid()
                y[..., 0:2] = (y[..., 0:2] * 2. - 0.5 + self.grid[i].to(x[i].device)) * self.stride[i]  # xy
                y[..., 2:4] = (y[..., 2:4] * 2) ** 2 * self.anchor_grid[i]  # wh
                z.append(y.view(bs, -1, self.no))

Please help.

**Comment by hghari** [comment]
The range of the numbers are different

**Comment by hghari** [comment]
not totally sure but I use this code for the output of onnx model:

stride=[8,16,32]
    anchor_grid=torch.tensor([[[[[[ 10.,  13.]]],
          [[[ 16.,  30.]]],
          [[[ 33.,  23.]]]]],
        [[[[[ 30.,  61.]]],
          [[[ 62.,  45.]]],
          [[[ 59., 119.]]]]],
        [[[[[116.,  90.]]],
          [[[156., 198.]]],
          [[[373., 326.]]]]]])
    z=[]
    for i in range(3):
        x=torch.from_numpy(res[keys[i]])
        a, b, c, d, e = np.shape(x)
        y=x.sigmoid()
        grid=_make_grid(nx=c,ny=d)
        y[..., 0:2] = (y[..., 0:2] * 2. - 0.5 + grid) * stride[i]
        y[..., 2:4] = (y[..., 2:4] * 2) ** 2 * anchor_grid[i]
        z.append(y.view(1, -1, e))
    ress=torch.cat(z, 1)

**Comment by m-wei** [comment]
> not totally sure but I use this code for the output of onnx model:
> 
> stride=[8,16,32]
> anchor_grid=torch.tensor([[[[[[ 10., 13.]]],
> [[[ 16., 30.]]],
> [[[ 33., 23.]]]]],
> [[[[[ 30., 61.]]],
> [[[ 62., 45.]]],
> [[[ 59., 119.]]]]],
> [[[[[116., 90.]]],
> [[[156., 198.]]],
> [[[373., 326.]]]]]])
> z=[]
> for i in range(3):
> x=torch.from_numpy(res[keys[i]])
> a, b, c, d, e = np.shape(x)
> y=x.sigmoid()
> grid=_make_grid(nx=c,ny=d)
> y[..., 0:2] = (y[..., 0:2] * 2. - 0.5 + grid) * stride[i]
> y[..., 2:4] = (y[..., 2:4] * 2) ** 2 * anchor_grid[i]
> z.append(y.view(1, -1, e))
> ress=torch.cat(z, 1)

hi,can you show the complete code?  what is res and res[keys]? Thanks!!

**Comment by hghari** [comment]
res is the result of the transformed output of onnx. keys are the names of the output motrixes
